platform/linux-generic/odp_packet_io.c

```diff
@@ -967,7 +1003,11 @@ static inline int packet_vector_send(odp_pktout_queue_t pktout_queue, odp_event_
 		ODP_ASSERT(entry != NULL);
 
 		odp_atomic_add_u64(&entry->s.stats_extra.out_discards, discards);
-		odp_packet_free_multi(&pkt_tbl[sent], discards);
+
+		if (odp_unlikely(_odp_pktio_tx_compl_enabled(entry)))
+			_odp_pktio_send_tx_compl_ev_pkt_multi(pkt_tbl, sent, num);
+		else
+			odp_packet_free_multi(&pkt_tbl[sent], discards);
```

Comment:
Code would be more readable if _odp_pktio_send_tx_compl_ev_pkt_multi() parameter would be just pkt[] and number of packets. Here it would be easy to see that tx_compl_ev_pkt_multi() and packet_free_multi() handle the same set of packets.